src/welearnbot/utils.py

```diff
@@ -111,6 +111,8 @@ def download_resource(
 ) -> Tuple[str, str]:
     """Helper function to retrieve a file/resource from the server"""
     filename = resource["filename"]
+    for i in range(len(subfolders)):
+        subfolders[i] = subfolders[i].strip()
```

Comment:
You are on the right track. But it's not a good idea to modify a list while you looping though it. I would suggest an alternative solution like this 👇🏻 
```py
subfolders = [subfolder.strip() for subfolder in subfolders]
```

Reply:
Alright that makes sense. Thanks. I have made the change.

Reply:
Looks good now, merging.